DuckDuckGo/TabSwitcherViewController.swift

```diff
@@ -114,12 +111,12 @@ class TabSwitcherViewController: UIViewController {
         delegate.tabSwitcher(self, didRemoveTabAt: index)
         collectionView.reloadData()
         refreshTitle()
-    }
-    
-    fileprivate func dismiss() {
         if tabsModel.isEmpty {
             delegate.tabSwitcherDidRequestClearAll(tabSwitcher: self)
         }
+    }
```

Comment:
This was a confusing part of the diff to read at first; but I think I get it now. You want this clear all called on onClear, and dismiss just does dismiss.